### PR TITLE
Dev: update gsoc idea2020

### PR DESCRIPTION
**Pull request by khancyr**
Remove multiples mention of jevois camera as I don't we us to be spam with jevois on proposal without having even read the camera capabilities.
Add some more project idea

### CODE REVIEW

**Comment by rajat2004** [comment]
Rover autotune, Improved ROS and Gazebo support, etc.! Would be great to see such projects! 

Just some ideas which I had, up to the devs and mentors, if it's actually something useful for the ArduPilot community, etc.  -

- Support for [MAVSDK](https://mavsdk.mavlink.io/develop/en/index.html)
- Additions to the AirSim simulator -
   - Support for more vehicle types such as boats, different models or types of rovers (similar to the idea for Morse)
   - Support for the full set of [AirSim APIs](https://github.com/microsoft/AirSim/blob/master/docs/apis.md)

**Comment by jmachuca77** [comment]
@rajat2004 I had already proposed MAVSDK support at least for swift. Not sure what the other languages of MAVSDK would need to make them work. That PR has not been merged yet https://github.com/ArduPilot/ardupilot_wiki/pull/2499

**Comment by rajat2004** [comment]
@jmachuca77 Hadn't seen that PR, edited the comment, discussion for MAVSDK would be better on the other PR then

File: dev/source/docs/gsoc-ideas-list.rst [hunk 1]
```diff
@@ -9,17 +9,20 @@ This is a list of projects suggested by ArduPilot developers for `GSoC 2020 <htt
 - :ref:`Non-GPS navigation improvements using Intel RealSense cameras <common-vio-tracking-camera>`
 - :ref:`Object avoidance <common-object-avoidance-landing-page>` improvements for Multicopters and/or Rovers
 - `Optical flow <https://ardupilot.org/copter/docs/common-optical-flow-sensors-landingpage.html>`__ position hold performance improvements for multicopters
-- `Precision Landing <https://ardupilot.org/copter/docs/precision-landing-with-irlock.html>`__ accuracy improvements using `JeVois camera <http://www.jevois.org/>`__ (or similar) for multicopters
-- Automatic docking for cars and boats using `JeVois camera <http://www.jevois.org/>`__ (or similar)
+- `Precision Landing <https://ardupilot.org/copter/docs/precision-landing-with-irlock.html>`__ accuracy improvements for multicopters
```
Comment (rmackay9) [inline dev/source/docs/gsoc-ideas-list.rst]: I'd prefer to keep this one as it was specifically talking about the jevois camera

File: dev/source/docs/gsoc-ideas-list.rst [hunk 2]
```diff
@@ -62,21 +65,35 @@ Precision Landing accuracy improvements for Multicopters
...
+- Improve the reliability and accuracy of the small EKF used to estimate the landing target's relative position and velocity. In particular adding protection against large lag in the readings leading to unreasonable results.
+- Add a landing state machine for landing retry and failure handling
 
 Lane following or visual follow-me for Copter, Rover or Boat
 ------------------------------------------------------------
 
 This project involves using machine vision and/or machine learning to add lane following or visual follow-me to ArduPilot's Copter or Rover firmware
 
-- Either a low-cost `JeVois camera <http://www.jevois.org/>`__ or a high powered `companion computer <https://ardupilot.org/dev/docs/companion-computers.html>`__ like the NVidia TX2 could be used
+- Either a low-cost camera or a high powered `companion computer <https://ardupilot.org/dev/docs/companion-computers.html>`__ like the NVidia TX2 could be used
```
Comment (rmackay9) [inline dev/source/docs/gsoc-ideas-list.rst]: As mentioned already I'd prefer to keep the specific mention of the jevois camera

File: dev/source/docs/gsoc-ideas-list.rst [hunk 3]
```diff
@@ -9,17 +9,20 @@ This is a list of projects suggested by ArduPilot developers for `GSoC 2020 <htt
...
 - Walking robot support
 - 3D aerobatic support for fixed wing aircraft
 - Improve :ref:`Morse simulator <sitl-with-morse>` integration including setup to move camera with vehicles
 - Create new vehicle models for the Morse simulator, including boats, planes and copters
+- Improve :ref:`Gazebo simulator <using-gazebo-simulator-with-sitl>` integration including json protocol, Gazebo9, and new sensors set
 - `MathWorks SimuLink <https://www.mathworks.com/products/simulink.html>`__ interface to ArduPilot SITL
-- `AirSim drone simulator <https://github.com/Microsoft/AirSim/>`__ support for ArduPilot SITL
 - Build system improvements, specifically fixing dependency handling and speeding up the waf build
-- Improvements to the `MAVProxy GCS <https://github.com/ArduPilot/MAVProxy>`__. Adding a parameter editor module, improving waypoint editor. Requires strong python skills.
+- Improvements to the `MAVProxy GCS <https://github.com/ArduPilot/MAVProxy>`__. Better multivehicle support, performance improvement. Requires strong python skills.
```
Comment (rmackay9) [inline dev/source/docs/gsoc-ideas-list.rst]: We should check these items for MAVProxy are what Tridge wants the focus to be on

File: dev/source/docs/gsoc-ideas-list.rst [hunk 4]
```diff
@@ -9,17 +9,20 @@ This is a list of projects suggested by ArduPilot developers for `GSoC 2020 <htt
 - :ref:`Non-GPS navigation improvements using Intel RealSense cameras <common-vio-tracking-camera>`
 - :ref:`Object avoidance <common-object-avoidance-landing-page>` improvements for Multicopters and/or Rovers
 - `Optical flow <https://ardupilot.org/copter/docs/common-optical-flow-sensors-landingpage.html>`__ position hold performance improvements for multicopters
-- `Precision Landing <https://ardupilot.org/copter/docs/precision-landing-with-irlock.html>`__ accuracy improvements using `JeVois camera <http://www.jevois.org/>`__ (or similar) for multicopters
-- Automatic docking for cars and boats using `JeVois camera <http://www.jevois.org/>`__ (or similar)
+- `Precision Landing <https://ardupilot.org/copter/docs/precision-landing-with-irlock.html>`__ accuracy improvements for multicopters
+- Automatic docking for cars and boats using camera
+- State estimation from camera or optical flow
```
Comment (rmackay9) [inline dev/source/docs/gsoc-ideas-list.rst]: @khancyr, are you looking to mentor these new items?

- State estimation from camera or optical flow
- Rover Autotune
- Unified performance counter on ArduPilot

File: dev/source/docs/gsoc-ideas-list.rst (hunk 1 above)
Comment (khancyr) [reply dev/source/docs/gsoc-ideas-list.rst]: done

File: dev/source/docs/gsoc-ideas-list.rst (hunk 4 above)
Comment (khancyr) [reply dev/source/docs/gsoc-ideas-list.rst]: I have remove the state estimation from optical flow, as I don't find a good reference paper to implement it

File: dev/source/docs/gsoc-ideas-list.rst (hunk 3 above)
Comment (khancyr) [reply dev/source/docs/gsoc-ideas-list.rst]: Parameter editor was already done last year, I just update with some stuff we speak lately.

**Pull request merged**